zebra/zebra_vty.c

```diff
@@ -985,9 +983,9 @@ vty_show_ip_route (struct vty *vty, struct route_node *rn, struct route_entry *r
 	}
       else
 	vty_out (vty, "  %c%*c",
-		 CHECK_FLAG (nexthop->flags, NEXTHOP_FLAG_FIB)
-		 ? '*' : ' ',
-		 len - 3 + (2 * recursing), ' ');
+	         CHECK_FLAG (nexthop->flags, NEXTHOP_FLAG_FIB)
+	         ? '*' : ' ',
+	         len - 3 + (2 * (nexthop->rparent ? 1 : 0)), ' ');
```

Comment:
If this was correct, it could be `nexthop->rparent ? 2 : 0`. However I think that something like `2 * nexthop_rec_level(nexthop)` might be better, with the function being a small helper counting the recursion level.